distributions/test/src/test/java/com/connexta/ion/replication/distributions/test/ReplicationITest.java

```diff
@@ -0,0 +1,144 @@
+package com.connexta.ion.replication.distributions.test;
+
+import static com.jayway.restassured.RestAssured.given;
+import static org.awaitility.Awaitility.await;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableMap;
+import com.jayway.restassured.path.xml.XmlPath;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+
+// TODO: why does this need to end in "test" for the tests to run?
+public class ReplicationITest extends IntegrationTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ReplicationITest.class);
+
+  private static final Slf4jLogConsumer logConsumer = new Slf4jLogConsumer(LOGGER);
+
+  private Set<String> idsCreated = new HashSet<>();
+
+  @BeforeClass
+  public static void beforeClass() {
+    replication.followOutput(logConsumer);
+    solrUrl = "http://" + solr.getContainerIpAddress() + ":" + solr.getFirstMappedPort();
+    sourceUrl = System.getenv("source.node");
+    destinationUrl = System.getenv("destination.node");
+    if (sourceUrl == null) {
+      sourceUrl = "https://ditto-1.phx.connexta.com:8993/services";
+    }
+    if (destinationUrl == null) {
+      destinationUrl = "https://ditto-2.phx.connexta.com:8993/services";
+    }
+  }
+
+  @After
+  public void cleanUp() {
+    idsCreated.forEach(this::deleteFromNodes);
+    clearAllSolrCores();
+  }
+
+  @Test
+  public void createXmlFile() throws Exception {
+    LOGGER.info("Solr URL: {}", solrUrl);
+    String testId = UUID.randomUUID().toString();
+
+    // create product
+    String id =
+        createProductOnSourceNode(
+            APPLICATION_XML, "data/reuters-xml-sample", ImmutableMap.of("testId", testId));
+    LOGGER.info("Returned ID: {}", id);
+    idsCreated.add(id);
+
+    // create nodes and jobs
+    postToSolrCore(
```

Comment:
❓ Can the configs be created in Before so that all of them can use the configs?

Reply:
Or at the minimum, the creation and waiting for it to be there should be done in one utility method that can then be reusable whenever one needs to populate a new entry for testing.

Reply:
I though a test may need to choose a specific sites or jobs file if it needs a particular set but for now creating them in a Before method should be fine.

Reply:
Personally, I wouldn't create it in `@Before` since the chance of using exactly that in the next test case is slim. I would, however, grouped together the logic into a utility method that you call from here. Or has done in codice-test for config admin, have a way to specify what should be pre-initialized for a given test case before that test case starts. That too me helps separate the pre-conditions (and the corresponding cleanup) from the actual test case

Reply:
I'm having some trouble figuring out how config admin in codice test works, but it seems like we would just be pushing that logic into an annotation instead of a method call? I think I'll stick with making a util method for now.